Assets/Scripts/UI/InGame/Graphics/GraphicsController.cs

```diff
@@ -11,91 +9,166 @@ public class GraphicsController : MonoBehaviour {
 		public GeneralGraphics GeneralGraphic;
 		public QualitySwitcher QualitySwitcher;
 		public ResolutionSwitcher ResolutionSwitcher;
+		public FPSLimiter FPSLimiter;
 		public Text label;
 		public Toggle CustomSettings;
 
+		// Index of the custom quality level
+		public const int CUSTOM = 6;
 
-		private void Update() {
-			
-			if(!CustomSettings.isOn)
-			{
-				GeneralGraphic.TextureQuality.interactable = false;
-				GeneralGraphic.ShadowRes.interactable = false;
-				GeneralGraphic.AntiAliasing.interactable = false;
-				GeneralGraphic.Shadows.interactable = false;
-				GeneralGraphic.VSync.interactable = false;
-				GeneralGraphic.SoftParticles.interactable = false;
 
-				QualitySwitcher.Slider.interactable = true;
-				
-				
+		private void Start()
+		{
+			if(PlayerPrefs.HasKey("GraphicsData") && PlayerPrefs.HasKey("QualityProfile") && PlayerPrefs.HasKey("FPSLimit"))
+			{
+				LoadGraphicPlayerPrefs();
+				Debug.Log("Has Prefs");
 			}
 			else
 			{
-				QualitySwitcher.Label.text = "Custom";
+				QualitySwitcher.Slider.value = QualitySettings.GetQualityLevel();
+				QualitySwitcher.UpdateQuality();
+				ChangeObjectValues();
 			}
 
+			AdvancedOptions();
+			
 		}
 
 		public void SaveGraphicPlayerPrefs()
 		{
-			// graphics
-			var data = new GeneralGraphics.Data(GeneralGraphic.TextureQuality.value, GeneralGraphic.ShadowRes.value, GeneralGraphic.AntiAliasing.value, GeneralGraphic.Shadows.value, GeneralGraphic.VSync.isOn, GeneralGraphic.SoftParticles.isOn, CustomSettings.isOn);
-			string json = JsonUtility.ToJson(data);
-			PlayerPrefs.SetString("GraphicsData", json);
+			try
+			{
+				var data1 = new GeneralGraphics.GraphicsData(GeneralGraphic);
+				string json1 = JsonUtility.ToJson(data1);
+				PlayerPrefs.SetString("GraphicsData", json1);
+
+				// quality profile
+				var data2 = new QualitySwitcher.QualityData(QualitySwitcher);
+				string json2 = JsonUtility.ToJson(data2);
+				PlayerPrefs.SetString("QualityProfile", json2);
 
-			// quality profile
-			var _data = new QualitySwitcher.Data((int)QualitySwitcher.Slider.value);
-			string _json = JsonUtility.ToJson(_data);
-			PlayerPrefs.SetString("QualityProfile", _json);
+				// fps limit
+				var data3 = new FPSLimiter.FPSData(FPSLimiter);
+				string json3 = JsonUtility.ToJson(data3);
+				PlayerPrefs.SetString("FPSLimit", json3);
 
-			PlayerPrefs.Save();	
+				PlayerPrefs.Save();
 
-			label.color = Color.green;
-			label.text = "saved player prefs";
+				label.color = Color.green;
+				label.text = "saved player prefs";
+			}
+			catch(Exception ex)
```

Comment:
Does it throw an exception here? If so, what is the expected exception? Using catch Exception should be avoided as this will not be logged.